File: src/types/worker.ts
/** 워커 데이터 인터페이스 */
export interface WorkerData {
  id: string;
  name: string;
  phone: string;
  residence: string;
  genderCode: number;
  positionCode: number;
  createDate: string;
  groupId?: string;
}

/** 워커 생성 API 바디 */
export type WorkerCreateBody = Pick<
  WorkerData,
  'name' | 'genderCode' | 'phone' | 'residence' | 'positionCode' | 'groupId'
>;

/** 워커 업데이트 API 바디 */
export type WorkerUpdateBody = Partial<Omit<WorkerCreateBody, 'genderCode'>>;

/** 워커 그룹 이동 API 바디 */
export type WorkerMoveGroupBody = {
  groupId?: string;
  workerIds: string[];
};

File: backend/models/worker.py
from sqlalchemy import Column, ForeignKey, Integer, String, DateTime
from db.base_class import Base
from datetime import datetime
from sqlalchemy.orm import relationship


class Worker(Base):
    __tablename__ = "worker"

    id = Column(Integer, primary_key=True, index=True)  # PK

    name = Column(String, nullable=False)  # 이름
    phone = Column(String, nullable=False)  # 연락처
    residence = Column(String, nullable=False)  # 거주지
    gender_code = Column(Integer, nullable=False)  # 성별 코드
    position_code = Column(Integer, nullable=False)  # 직위 코드

    create_date = Column(DateTime(timezone=True), nullable=False, default=datetime.now)

    group_id = Column(Integer, ForeignKey("group.id"), nullable=True)
    group = relationship("Group", back_populates="worker")

    personal = relationship(
        "Personal",
        back_populates="worker",
        cascade="all, delete-orphan",
    )
    contract = relationship("Contract", uselist=False, back_populates="worker")
    worklogs = relationship(
        "WorkLog", back_populates="worker", cascade="all, delete-orphan"
    )
